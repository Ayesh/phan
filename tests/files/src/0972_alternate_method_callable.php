<?php

namespace NS972;
class Foo {
    public function m(int $arg): void {
        echo "In Foo::m\n";
    }
}
class Bar extends Foo {
    public function m(int $arg): void {
        echo "In Bar::m\n";
    }
}
// Note that phan has never implemented resolution for this callable method name syntax.
call_user_func([new Foo(), 'm']);
call_user_func([new Foo(), 'NS972\Foo::m']);
call_user_func([new Foo(), 'NS972\Bar::m']);
call_user_func([new Bar(), 'm']);
call_user_func([new Bar(), 'NS972\Foo::m']);
call_user_func([new Bar(), 'NS972\Bar::m']);
call_user_func('NS972\Bar::NS972\Foo::m');  // invalid
(new Foo())->{'NS972\Foo::m'}();
