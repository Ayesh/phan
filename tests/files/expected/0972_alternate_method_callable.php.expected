%s:15 PhanParamTooFewCallable Call with 0 arg(s) to \NS972\Foo::m(int $arg) (as a provided callable) which requires 1 arg(s) defined at %s:5
%s:16 PhanDeprecatedPartiallySupportedCallableAlternateScope Saw partially supported callable [new Foo(), 'NS972\\Foo::m']. This was deprecated in PHP 8.2. Invoking methods with alternative scopes can be done with ReflectionMethod::invoke or Closure::bindTo instead.
%s:16 PhanUndeclaredMethodInCallable Call to undeclared method NS972\Foo::m in callable. Possible object type(s) for that method are \NS972\Foo
%s:17 PhanDeprecatedPartiallySupportedCallableAlternateScope Saw partially supported callable [new Foo(), 'NS972\\Bar::m']. This was deprecated in PHP 8.2. Invoking methods with alternative scopes can be done with ReflectionMethod::invoke or Closure::bindTo instead.
%s:17 PhanUndeclaredMethodInCallable Call to undeclared method NS972\Bar::m in callable. Possible object type(s) for that method are \NS972\Foo
%s:18 PhanParamTooFewCallable Call with 0 arg(s) to \NS972\Bar::m(int $arg) (as a provided callable) which requires 1 arg(s) defined at %s:10
%s:19 PhanDeprecatedPartiallySupportedCallableAlternateScope Saw partially supported callable [new Bar(), 'NS972\\Foo::m']. This was deprecated in PHP 8.2. Invoking methods with alternative scopes can be done with ReflectionMethod::invoke or Closure::bindTo instead.
%s:19 PhanUndeclaredMethodInCallable Call to undeclared method NS972\Foo::m in callable. Possible object type(s) for that method are \NS972\Bar
%s:20 PhanDeprecatedPartiallySupportedCallableAlternateScope Saw partially supported callable [new Bar(), 'NS972\\Bar::m']. This was deprecated in PHP 8.2. Invoking methods with alternative scopes can be done with ReflectionMethod::invoke or Closure::bindTo instead.
%s:20 PhanUndeclaredMethodInCallable Call to undeclared method NS972\Bar::m in callable. Possible object type(s) for that method are \NS972\Bar
%s:21 PhanUndeclaredStaticMethodInCallable Reference to undeclared static method \NS972\Bar::NS972\Foo::m in callable
%s:22 PhanUndeclaredMethod Call to undeclared method \NS972\Foo::NS972\Foo::m
